data/phones/list_phones.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+"""This script prints a tally of the phones/phonemes of a WikiPron TSV file.
+
+For each phone/phoneme, this script prints:
+- the phone/phoneme
+- the number of words that have this phone/phoneme
+- a few example word-pronunciation pairs for this phone/phoneme
+"""
+
+import argparse
+import collections
+import random
+
+from typing import Dict, List, Set
+
+
+def _get_cli_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("filepath", help="Path to TSV scraped by WikiPron")
+    args = parser.parse_args()
+    return args
+
+
+def _count_phones(filepath: str) -> Dict[str, Set[str]]:
+    # phone_to_examples as Dict[str, Set[str]] is the most straightforward
+    # data structure for our purposes. It's not memory-efficient
+    # (with the same word-pron pair appearing in different phones' sets),
+    # but anything fancier doesn't seem worth the work.
+    phone_to_examples = collections.defaultdict(set)
+    with open(filepath, encoding="utf-8") as f:
+        for line in f:
+            line = line.strip()
+            if not line:
+                continue
+            word, pron = line.split("\t", maxsplit=1)
+            example = f"( {word} | {pron} )"
+            phones = pron.split()
+            for phone in phones:
+                phone_to_examples[phone].add(example)
+    return phone_to_examples
+
+
+def _pick_examples_for_display(examples: Set[str]) -> List[str]:
+    # We could have exposed the maximum number of examples to display
+    # (set to be 3 now) to the command-line interface,
+    # but it doesn't seem worth it for the time being.
+    n_examples = min(len(examples), 3)
+    # Using list() here because Python 3.9 has deprecated the use
+    # of an _unordered_ set as the input to random.sample.
+    sample = random.sample(list(examples), n_examples)
+    return sample
+
+
+def main():
+    args = _get_cli_args()
+    phone_to_examples: Dict[str, Set[str]] = _count_phones(args.filepath)
+    for phone, examples in sorted(
+        phone_to_examples.items(), key=lambda x: len(x[1]), reverse=True
+    ):
+        print(
+            f"{phone:<5}"
+            f"{len(examples):<10}"
+            f"{', '.join(_pick_examples_for_display(examples))}"
+        )
+    print(f"\n# unique phones: {len(phone_to_examples)}")
+
+
+if __name__ == "__main__":
```

Comment:
I sort of prefer the style where argparsing goes here and is passed to main (I also don't think it has any good reason to be its own function). I believe it's what we've used with other tools, or it was what I used initially.

As is `main` is not reusable except in a command-line environment, which is the purpose of having `main` be its own function, is it not?